frontend-react/src/hooks/UseOrganizationPublicKeys.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback } from "react";
+
+import {
+    RSApiKeysResponse,
+    servicesEndpoints,
+} from "../config/endpoints/settings";
+import { useAuthorizedFetch } from "../contexts/AuthorizedFetchContext";
+import { useSessionContext } from "../contexts/SessionContext";
+
+const { publicKeys } = servicesEndpoints;
+
+export type UseOrganizationPublicKeysResult = {
+    orgPublicKeys: RSApiKeysResponse;
+    isLoading: boolean;
+};
+
+export const useOrganizationPublicKeys = () => {
+    const { authorizedFetch, rsUseQuery } =
+        useAuthorizedFetch<RSApiKeysResponse>();
+
+    const { activeMembership } = useSessionContext();
+    const memoizedDataFetch = useCallback(
+        () =>
+            authorizedFetch(publicKeys, {
+                segments: {
+                    orgName: activeMembership?.parsedName!!,
+                },
+            }),
+        [activeMembership?.parsedName, authorizedFetch]
+    );
+    const { data, isLoading } = rsUseQuery(
+        [publicKeys.queryKey, activeMembership],
+        memoizedDataFetch,
+        {
+            enabled:
+                !!activeMembership?.parsedName && !!activeMembership.service,
+        }
+    );
+
+    return { orgPublicKeys: data, isLoading };
```

Comment:
Our custom react-query hooks should just be a loose wrapper of configuration, and they should just pass back what `useQuery`/`useMutation` returns verbatim:

```ts
const query = rsUseQuery(...);

// or more simply
return rsUseQuery(...);
```

Destructuring beforehand limits the utility of the return value since we have to explicitly pass back `isLoading`, `isInitialLoading`, `status`, et cetera if we ever need that from the query/mutation.